src/catchup/CatchupManagerImpl.cpp

```diff
@@ -79,59 +88,78 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
     // First: if CatchupWork has started, just buffer and return early.
     if (mCatchupWork)
     {
+        // we don't want to buffer any ledgers at or below where we are catching
+        // up to
+        auto const& config = mCatchupWork->getCatchupConfiguration();
+        if (ledgerData.getLedgerSeq() <= config.toLedger())
+        {
+            return;
+        }
+
         addToSyncingLedgers(ledgerData);
         logAndUpdateCatchupStatus(true);
         return;
     }
 
-    // Next switch between 3 cases: starting buffering, waiting to
-    // buffer, and already buffering.
+    // Next, we buffer every out of sync ledger to allow us to get back in sync
+    // in case the ledgers we're missing are received.
+    addToSyncingLedgers(ledgerData);
+
+    // Keep track of the smallest buffered checkpoint.
```

Comment:
I don't see the point in dealing with the complexity of managing this member `mSmallestBufferedCheckpoint `: you can easily compute it given the map of buffered ledgers (just like how you compute the "highest checkpoint" in `trimSyncingLedgers`).

Reply:
Keeping track of the smallest checkpoint was done because it was more efficient than looking through the buffered ledgers, but the buffered ledgers should never grow that large before catchup starts, so a search should be fine. I'll remove `mSmallestBufferedCheckpoint`.

Reply:
Added a `findFirstCheckpoint` method to use here and in `trimSyncingLedgers`.